src/util-file.c

```diff
@@ -454,6 +454,14 @@ static File *FileAlloc(const uint8_t *name, uint16_t name_len)
     new->name_len = name_len;
     memcpy(new->name, name, name_len);
 
+    new->sid_cnt = 0;
+    new->sid_max = 8;
+    new->sid = SCMalloc(sizeof(uint32_t) * new->sid_max);
+    if (new->sid == NULL) {
+        SCFree(new);
+        return NULL;
```

Comment:
I'm wondering if this should lead to a failure on the whole file. It's not essential for matching and file tracking, so perhaps we need to allow for this to fail. It's extremely unlikely to happen of course, but not impossible.